src/qoi/src/absorption_coeff.C

```diff
@@ -0,0 +1,290 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// GRINS - General Reacting Incompressible Navier-Stokes
+//
+// Copyright (C) 2014-2016 Paul T. Bauman, Roy H. Stogner
+// Copyright (C) 2010-2013 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+
+// This class
+#include "grins/absorption_coeff.h"
+#include "grins/variable_warehouse.h"
+#include "grins/physical_constants.h"
+#include "grins/math_constants.h"
+
+#if GRINS_HAVE_ANTIOCH
+#include "grins/antioch_chemistry.h"
+#endif
+
+#if GRINS_HAVE_CANTERA
+#include "grins/cantera_mixture.h"
+#endif
+
+namespace GRINS
+{
+  template<typename Chemistry>
+  AbsorptionCoeff<Chemistry>::AbsorptionCoeff(SharedPtr<Chemistry> & chem, SharedPtr<HITRAN> & hitran,
+                                              libMesh::Real nu_min, libMesh::Real nu_max,
+                                              libMesh::Real desired_nu, const std::string & species,
+                                              libMesh::Real thermo_pressure)
+    : _chemistry(chem),
+      _hitran(hitran),
+      _nu(desired_nu),
+      _T_var(GRINSPrivate::VariableWarehouse::get_variable_subclass<PrimitiveTempFEVariables>("Temperature")),
+      _P_var(GRINSPrivate::VariableWarehouse::get_variable_subclass<PressureFEVariable>("Pressure")),
+      _Y_var(GRINSPrivate::VariableWarehouse::get_variable_subclass<SpeciesMassFractionsVariable>("SpeciesMassFractions")),
+      _T0(296), // [K]
+      _Pref(1), // [atm]
+      _rad_coeff(1.43887752) // [cm K]
+  {
+    // sanity checks
+    libmesh_assert_greater(nu_max,nu_min);
```

Comment:
I would actually make these runtime checks/errors, not just debug/devel mode checks.